Utility/functions.cs

```diff
@@ -260,47 +252,62 @@ public static void SnapshotItem(ItemDrop item, float lightIntensity = 1.3f, Quat
             topLight.cullingMask = 1 << layer;
             topLight.intensity = lightIntensity;
 
-            Rect rect = new(0, 0, 64, 64);
-
-            GameObject visual = UnityEngine.Object.Instantiate(item.transform.Find("attach").gameObject);
-            foreach (Transform child in visual.GetComponentsInChildren<Transform>())
+            try
             {
-                child.gameObject.layer = layer;
-            }
+                Rect rect = new(0, 0, 64, 64);
 
-            Renderer[] renderers = visual.GetComponentsInChildren<Renderer>();
-            Vector3 min = renderers.Aggregate(Vector3.positiveInfinity, (cur, renderer) => renderer is ParticleSystemRenderer ? cur : Vector3.Min(cur, renderer.bounds.min));
-            Vector3 max = renderers.Aggregate(Vector3.negativeInfinity, (cur, renderer) => renderer is ParticleSystemRenderer ? cur : Vector3.Max(cur, renderer.bounds.max));
-            Vector3 size = max - min;
+                Transform target = item.transform.Find("attach"); // ?? item.transform.Find("attach_skin");
 
-            camera.targetTexture = RenderTexture.GetTemporary((int)rect.width, (int)rect.height);
-            float maxDim = Mathf.Max(size.x, size.z);
-            float minDim = Mathf.Min(size.x, size.z);
-            float yDist = (maxDim + minDim) / Mathf.Sqrt(2) / Mathf.Tan(camera.fieldOfView * Mathf.Deg2Rad);
-            Transform transform = camera.transform;
-            transform.position = ((min + max) / 2) with { y = max.y } + new Vector3(0, yDist, 0);
-            topLight.transform.position = transform.position + new Vector3(-2, 0, 0.2f) / 3 * -yDist;
+                if (!target)
```

Comment:
Added fallback game object in the event that 'attach' doesn't exist (in some cases it doesn't).